thirdeye/thirdeye-frontend/app/utils/yaml-tools.js

```diff
@@ -29,6 +28,20 @@ rules:
     type: DATA_SLA              # Alert if data is missing.
     params:
       sla: 3_DAYS               # Data is missing for 3 days since last availability
+
+# You can mention a cron to allow this detection to be run periodically
+# For example, the cron below would execute every 5 minutes.
+# cron: "0 0/5 * 1/1 * ? *"
+
+# Backfill detections
```

Comment:
The term backfill is a bit overloaded. It could also mean rerunning the detection which is not what is intended.

Reply:
Yes. That is a side effect but it happens today anyway. Whenever, you create an alert, you run the detection for the past month every time. This parameter just exposes the ability to configure the same.

But I totally agree on having separate atomic APIs in the longer term for creating alerts, doing detections and saving them. Currently the preview runs detections but does not save them. Creating an alert also fires detections but doesn't allow configuring the detection. Also, it doesn't seem to be possible to create a new alert without triggering a detectionjob.

Reply:
There are apis for each one of them. I will consider documenting it.